c10/core/impl/PyInterpreter.h

```diff
@@ -30,6 +30,46 @@ using Stack = std::vector<c10::IValue>;
 namespace c10 {
 namespace impl {
 
+struct C10_API PyInterpreter;
+
+struct C10_API CUDATraceFunctionWrapper {
+  using event_creation_sig = void(const PyInterpreter*, uintptr_t event);
+  using event_deletion_sig = void(const PyInterpreter*, uintptr_t event);
+  using event_record_sig =
+      void(const PyInterpreter*, uintptr_t event, uintptr_t stream);
+  using event_wait_sig =
+      void(const PyInterpreter*, uintptr_t event, uintptr_t stream);
+  using memory_allocation_sig = void(const PyInterpreter*, uintptr_t pointer);
+  using memory_deallocation_sig = void(const PyInterpreter*, uintptr_t pointer);
+  using stream_creation_sig = void(const PyInterpreter*, uintptr_t stream);
```

Comment:
There was a comment earlier about replacing `uintptr_t` with either `void*` or `CUevent_st*`/`CUstream_st*`/... (by forward-declaring these structs). I take it you've opted to maintain these as integers? That's perfectly fine, since they will be integers in Python eventually anyways, just wanted to check.

(Note that I'm not even sure anymore that using `CUevent_st`/... is doable, because of the hipification issues).

Reply:
ah yes, HIPification would mean we would have to give these non-standard names

Reply:
Yes, I thought I might leave them as `uintptr_t`.